well_known.go

```diff
@@ -79,10 +89,20 @@ func LookupWellKnown(serverNameType ServerName) (*WellKnownResult, error) {
 		}
 	}
 
-	body, err := ioutil.ReadAll(resp.Body)
+	// By this point we hope that we've caught any huge well-known records
+	// by checking Content-Length, but it's possible that header will be
+	// missing. Better to be safe than sorry by reading no more than the
+	// WellKnownMaxSize in any case.
+	bodyBuffer := make([]byte, WellKnownMaxSize)
+	limitedReader := &io.LimitedReader{
+		R: resp.Body,
```

Comment:
We still need to close the body at some point.

Reply:
There's a `defer` further up the function:
```
	defer func() {
		_ = resp.Body.Close()
	}()
```